materialize-sql/std_sql.go

```diff
@@ -46,12 +46,14 @@ func StdSQLExecStatements(ctx context.Context, db *sql.DB, statements []string)
 	// through a single connection. This allows a driver to explicitly run
 	// `BEGIN;` and `COMMIT;` statements around a transactional operation.
 	var conn, err = db.Conn(ctx)
-	if err == nil {
-		err = conn.PingContext(ctx)
-	}
 	if err != nil {
 		return fmt.Errorf("connecting to DB: %w", err)
 	}
+	defer conn.Close()
+
+	if err = conn.PingContext(ctx); err != nil {
+		return fmt.Errorf("connecting to DB: %w", err)
```

Comment:
Might help to have a different error message, so we can tell where the errors are coming from.